adflow/pyADflow.py

```diff
@@ -4135,6 +4178,26 @@ def computeJacobianVectorProductBwd(
         # Single return (most frequent) is 'clean', otherwise a tuple.
         return tuple(returns) if len(returns) > 1 else returns[0]
 
+    def computeJacobianVectorProductBwdFast(
+        self,
+        resBar=None,
+    ):
+        """used for testing the routines used for jac vec prods for the adjoint solve.
+
+        Parameters
+        ----------
+        resBar : numpy array
+            Seed for the residuals (dwb in adflow)
+
+        Returns
+        -------
+        wbar: array
+            state derivative seeds
+        """
+        wbar = self.adflow.adjointapi.computematrixfreeproductbwdfast(resBar, self.getAdjointStateSize())
```

Comment:
I am guessing there is a prerequisite that the intermediate variables need to be up to date before calling this? Can you provide a small guide type documentation on how to use these routines for debugging?

Reply:
The intermediate variables do need to be up to date. 
The tests provide an example of how to do this (just call get residuals before).

When this gets accepted I'll update [these docs](https://mdolab-private-docs.readthedocs-hosted.com/en/latest/tutorials/debugging.html#debugging-derivatives)

Reply:
Sounds good